From e63f93a970b8eb83d83dfb8bb39d578229fa3496 Mon Sep 17 00:00:00 2001
From: Andreas Arnez <arnez@linux.ibm.com>
Date: Tue, 14 May 2019 17:19:34 +0200
Subject: [PATCH] s390x: Clean up s390-check-opcodes.pl

Fix false positives when invoking s390-check-opcodes.pl.  Also clean up
some code formatting issues in that script.  Add the instructions TPEI and
IRBM to guest_s390_toIR.c and s390-opcodes.csv, so they are not longer
warned about.
---
 VEX/priv/guest_s390_toIR.c      |   2 +
 auxprogs/s390-check-opcodes.pl  | 101 +++++++++++++++++---------------
 docs/internals/s390-opcodes.csv |   2 +
 3 files changed, 57 insertions(+), 48 deletions(-)

Index: valgrind-3.15.0/VEX/priv/guest_s390_toIR.c
===================================================================
--- valgrind-3.15.0.orig/VEX/priv/guest_s390_toIR.c
+++ valgrind-3.15.0/VEX/priv/guest_s390_toIR.c
@@ -19615,8 +19615,10 @@ s390_decode_4byte_and_irgen(const UChar
    case 0xb99d: /* ESEA */ goto unimplemented;
    case 0xb99e: /* PTI */ goto unimplemented;
    case 0xb99f: /* SSAIR */ goto unimplemented;
+   case 0xb9a1: /* TPEI */ goto unimplemented;
    case 0xb9a2: /* PTF */ goto unimplemented;
    case 0xb9aa: /* LPTEA */ goto unimplemented;
+   case 0xb9ac: /* IRBM */ goto unimplemented;
    case 0xb9ae: /* RRBM */ goto unimplemented;
    case 0xb9af: /* PFMF */ goto unimplemented;
    case 0xb9b0: s390_format_RRF_M0RERE(s390_irgen_CU14, RRF3_r3(ovl),
